src/middlewared/middlewared/plugins/ldap.py

```diff
@@ -493,6 +580,19 @@ def get_dn(self, dn=None, ldap=None):
     async def __set_state(self, state):
         await self.middleware.call('cache.put', 'LDAP_State', state.name)
 
+    @private
+    def get_nslcd_status(self):
+        """
+        Returns internal nslcd state. nslcd will preferentially use the first LDAP server,
+        and only failover if the current LDAP server is unreachable.
+        """
+        ctx = NslcdClient(NlscdConst.NSLCD_ACTION_STATE_GET.value)
```

Comment:
Not important, but it seems a context manager for NslcdClient would be a perfect fit for this use case : P

Reply:
Good idea. I'll do that.